### PR TITLE
Use action client (instead of goal handle) to get result

### PR DESCRIPTION
**Pull request by orduno**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #923 |
| Primary OS tested on | Ubuntu 18.04 |
| Robotic platform tested on | Gazebo simulation of TB3 |

---

## Description of contribution in a few bullet points

Addressing #923.

### CODE REVIEW

File: nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp [hunk 1]
```diff
@@ -104,12 +104,8 @@ class BtActionNode : public BT::CoroActionNode
   {
     on_tick();
 
-    // Enable result awareness by providing an empty lambda function
-    auto send_goal_options = typename rclcpp_action::Client<ActionT>::SendGoalOptions();
```
[the remark this answers is not in the document]
Comment (crdelsey) [reply nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp]: @SteveMacenski Do the empty lambas prevent timeouts somehow, or are you thinking about using the action feedback to know that the action is still making progress?

If the latter, can we just add the functions back in when they actually exist and do something?

**Comment by SteveMacenski** [comment]
Any reason this isn't ready to merge?

**Comment by orduno** [comment]
> @SteveMacenski Do the empty lambas prevent timeouts somehow, or are you thinking about using the action feedback to know that the action is still making progress?
> If the latter, can we just add the functions back in when they actually exist and do something?

**Comment by SteveMacenski** [comment]
>  Do the empty lambas prevent timeouts somehow, or are you thinking about using the action feedback to know that the action is still making progress?

I don't know, I didn't write this. The impression I had is that that empty lambda was there for a reason, it was never filled out.

**Comment by orduno** [comment]
I think we only need the lambdas to make the action result [aware](https://github.com/ros2/rclcpp/blob/b92db52bb14ec097946bfe44d2b337359dddfbab/rclcpp_action/include/rclcpp_action/client.hpp#L380) when requesting through the goal handle.

That's not required when getting the result through the [client](https://github.com/ros2/rclcpp/blob/b92db52bb14ec097946bfe44d2b337359dddfbab/rclcpp_action/include/rclcpp_action/client.hpp#L410).

However, on your comment, you mention updates:
> we do need to keep these callbacks so we can have updates to stop having this timeout loop

Are there other side effects (beyond getting the result) from removing the callback on timeouts or feedback?

FYI, there is a separate callback for the [feedback](https://github.com/ros2/rclcpp/blob/b92db52bb14ec097946bfe44d2b337359dddfbab/rclcpp_action/include/rclcpp_action/client.hpp#L280).

**Comment by SteveMacenski** [comment]
Assuming this doesn’t break anything, I think its fine to merge.

**Pull request merged**